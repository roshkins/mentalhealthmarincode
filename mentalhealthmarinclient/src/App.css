.App {
  display: flex;
  flex-direction: column;
  height: 100%;
  margin: 0 2%;
}

.Providers {
  display: inline-block;
  /*background-color: hsla(90, 40%, 85%, 0.8);*/
  /*border-top: 1px solid hsla(90, 40%, 75%, 0.8);
  border-left: 1px solid hsla(90, 40%, 75%, 0.8);
  border-right: 1px solid hsla(90, 40%, 60%, 0.8);
  border-bottom: 1px solid hsla(90, 40%, 60%, 0.8);*/
  overflow-y: auto;
}
.Providers::-webkit-scrollbar {
  width: 2%;
}
.Providers::-webkit-scrollbar-track {
  background: hsla(90, 30%, 60%, 1);
}
.Providers::-webkit-scrollbar-thumb {
  background: linear-gradient(
    to right,
    hsla(90, 30%, 50%, 1) 10%,
    hsla(90, 30%, 40%, 1) 80%,
    hsla(90, 30%, 50%, 1)
  );
}

.NewProvider {
  padding: 2% 0;
}
.Provider {
  position: relative;
  box-sizing: border-box;
  display: inline-block;
  width: 20em;
  height: 15em;
  overflow: hidden;
  vertical-align: top;
  margin: 0 1em 1em 0;
  background: hsla(90, 40%, 85%, 0.8);
  border: 1px solid hsla(90, 40%, 60%, 0.8);
}
.ProviderCommon,
.ProviderFormal {
  text-shadow: none;
}

h1,
h2,
h3,
h4,
h5,
h6 {
  margin: 0.25em 0;
  font-family: 'Comfortaa', cursive;
  text-shadow: -3px -3px 3px hsla(90, 40%, 75%, 0.6),
    3px -3px 3px hsla(90, 40%, 75%, 0.6), -3px 3px 3px hsla(90, 40%, 75%, 0.6),
    3px 3px 3px hsla(90, 40%, 75%, 0.6);
}

.NewProvider input,
.NewProvider textarea {
  width: 25em;
  box-sizing: border-box;
  background-color: hsla(90, 40%, 85%, 0.8);
  border: 0;
  font-family: 'Mukta Mahee', sans-serif;
  margin: 0.1em;
}

.NewProvider input[type="submit"] {
  cursor: pointer;
}

.Title {
  padding: 2em 0;
}
