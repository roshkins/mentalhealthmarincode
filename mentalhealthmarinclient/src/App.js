import React, { Component } from "react";
import Providers from "./containers/providers";
import NewProvider from "./containers/new-provider";
import "./App.css";
import Landscape from "./landscape.jpg";
class App extends Component {
  constructor(params) {
    super(params);
    this.state = {
      providers: [],
      newProvider: {
        commonName: "",
        formalName: "",
        usageInstructions: ""
      }
    };
  }

  render() {
    return (
      <div className="App">
        <h1 className="Title">Here is a list of places you can seek help!</h1>
        <Providers />
        <NewProvider />
      </div>
    );
  }
}

export default App;
