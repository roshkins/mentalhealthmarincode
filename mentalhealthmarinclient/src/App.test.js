import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import App from "./App";

//Redux stuff
import { Provider } from "react-redux";
import { createStore } from "redux";
import AppReducer from "./reducers/app";

//https://github.com/facebookincubator/create-react-app/issues/967#issuecomment-256358738
beforeEach(function() {
  global.fetch = jest.fn().mockImplementation(() => {
    var p = new Promise((resolve, reject) => {
      resolve({
        ok: true,
        Id: "123",
        json: function() {
          return [
            {
              formal_name: "string",
              common_name: "string",
              phone_number: "string",
              email_address: "string",
              address: "string",
              event_calendar: "string",
              usage_instructions: "string",
              id: 0
            }
          ];
        }
      });
    });

    return p;
  });
});

it("renders without crashing", () => {
  let store = createStore(AppReducer);

  const div = document.createElement("div");
  ReactDOM.render(
    <Provider store={store}>
      <App />
    </Provider>,
    div
  );
});

it("looks the same as before", () => {
  let store = createStore(AppReducer);

  const tree = renderer
    .create(
      <Provider store={store}>
        <App />
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
