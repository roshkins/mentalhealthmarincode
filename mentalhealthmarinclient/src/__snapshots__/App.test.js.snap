// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`looks the same as before 1`] = `
<div
  className="App"
>
  <h1>
    Here is a list of places you can seek help!
  </h1>
  <div>
    <h1>
       Add your own! 
    </h1>
    <form
      onSubmit={[Function]}
    >
      <input
        onChange={[Function]}
        placeholder="common name"
        type="text"
        value=""
      />
      <br />
      <input
        onChange={[Function]}
        placeholder="email address for provider"
        type="email"
        value=""
      />
      <br />
      <input
        onChange={[Function]}
        placeholder="formal name"
        type="text"
        value=""
      />
      <br />
      <textarea
        onChange={[Function]}
        placeholder="usage instructions"
        value=""
      />
      <br />
      <input
        type="submit"
        value="Add your own!"
      />
    </form>
  </div>
</div>
`;
